src/Microsoft.DotNet.Interactive.CSharpProject/BuildCacheFileUtilities.cs

```diff
@@ -0,0 +1,146 @@
+                <Type>String</Type>
+              </ProjectData>
+              <ProjectData Include=""PropertyLangVersion=$(LangVersion)"">
+                <Type>String</Type>
+              </ProjectData>
+              <ProjectData Include=""PropertyOutputType=$(OutputType)"">
+                <Type>String</Type>
+              </ProjectData>
+            </ItemGroup>
+        
+            <!-- Split PreprocessorSymbols into individual items -->
+            <ItemGroup>
+              <PreprocessorSymbolItems Include=""$(DefineConstants.Split(';'))"" />
+            </ItemGroup>
+        
+            <!-- Transform the ProjectData and PreprocessorSymbolItems to include the prefix -->
+            <ItemGroup>
+              <ProjectDataLines Include=""$([System.String]::Format('{{0}}{{1}}', %(ProjectData.Prefix), %(ProjectData.Identity)))"" />
+              <ProjectDataLines Include=""$([System.String]::Format('PreprocessorSymbols={{0}}', %(PreprocessorSymbolItems.Identity)))"" />
+            </ItemGroup>
+        
+            <!-- Write collected project data to a file -->
+            <WriteLinesToFile Lines=""@(ProjectDataLines)""
+                              File=""$(MSBuildProjectFullPath).interactive.workspaceData.cache""
+                              Overwrite=""True""
+                              WriteOnlyWhenDifferent=""True"" />
+          </Target>
+        </Project>";
+
+    public record BuildProjectData
+    {
+        public Guid ProjectGuid { get; init; }
+        public IEnumerable<string> ProjectReferences { get; init; }
+        public string ProjectFilePath { get; init; }
+        public string LanguageName { get; init; }
+        public string PropertyTargetPath { get; init; }
+        public string[] SourceFiles { get; init; }
+        public string[] References { get; init; }
+        public string[] AnalyzerReferences { get; init; }
+        public string[] PreprocessorSymbols { get; init; }
+        public string PropertyLangVersion { get; init; }
+        public string PropertyOutputType { get; init; }
+        // TODO : Modify the target to get this value
+        public string PropertyDefineConstants { get; init; }
+        public string[] CompileInputs { get; init; }
+    }
+
+    internal static async Task<string> BuildAndCreateCacheFileAsync(string csprojFilePath)
+    {
+        DirectoryInfo directoryInfo = new DirectoryInfo(Path.GetDirectoryName(csprojFilePath));
+        FileInfo lastBuildErrorLogFile = new FileInfo(Path.Combine(directoryInfo.FullName, ".trydotnet-builderror"));
+
+        CleanObjFolder(directoryInfo);
+
+        string tempDirectoryBuildTarget =
+            Path.Combine(Path.GetDirectoryName(csprojFilePath), BuildCacheFileUtilities.DirectoryBuildTargetFilename);
+        File.WriteAllText(tempDirectoryBuildTarget, BuildCacheFileUtilities.DirectoryBuildTargetsContent);
+
+        var args = "";
+        if (Path.Exists(csprojFilePath) == true)
```

Comment:
```suggestion
        if (Path.Exists(csprojFilePath))
```

Reply:
Replacing '==' with 'is' for null check in the codebase for this project